src/CAN/FakeCANBoard.cpp

```diff
@@ -36,11 +36,20 @@ std::unordered_set<int> modes = {
 	static_cast<int>(TestMode::LimitSwitch),   static_cast<int>(TestMode::ScienceTelemetry),
 	static_cast<int>(TestMode::ScienceMotors), static_cast<int>(TestMode::ScienceServos)};
 
-int prompt(const char* msg) {
+int prompt(std::string_view message) {
 	std::string str;
-	std::cout << msg << " > ";
-	std::getline(std::cin, str);
-	int val = std::stoi(str, nullptr, 0);
+	int val;
+	bool valid_input = false;
+	do {
+		std::cout << message << " > ";
+		std::getline(std::cin, str);
+		try {
+			val = std::stoi(str, nullptr, 0);
+			valid_input = true;
+		} catch (const std::exception&) {
```

Comment:
Catching all exceptions is generally a code smell. Can we explicitly catch the errors we care about, i.e. `std::invalid_argument` and `std::out_of_range`?